platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/pool/TransactionPoolNexus.java

```diff
@@ -76,13 +109,116 @@ public class TransactionPoolNexus implements TransactionSupplier {
     public TransactionPoolNexus(@NonNull final PlatformContext platformContext) {
         Objects.requireNonNull(platformContext);
 
+        illegalTransactionLogger = new RateLimitedLogger(logger, platformContext.getTime(), Duration.ofMinutes(10));
+
         final TransactionConfig transactionConfig =
                 platformContext.getConfiguration().getConfigData(TransactionConfig.class);
         maxTransactionBytesPerEvent = transactionConfig.maxTransactionBytesPerEvent();
         throttleTransactionQueueSize = transactionConfig.throttleTransactionQueueSize();
 
         transactionPoolMetrics = new TransactionPoolMetrics(
                 platformContext, this::getBufferedTransactionCount, this::getPriorityBufferedTransactionCount);
+
+        maximumTransactionSize = transactionConfig.transactionMaxBytes();
+
+        final EventCreationConfig eventCreationConfig =
+                platformContext.getConfiguration().getConfigData(EventCreationConfig.class);
+        maximumPermissibleUnhealthyDuration = eventCreationConfig.maximumPermissibleUnhealthyDuration();
+    }
+
+    // FUTURE WORK: these checks should be unified with the checks performed when a system transaction is submitted.
+    // The reason why this method coexists with submitTransaction() is due to legacy reasons, not because it
+    // actually makes sense to have this distinction.
+
+    /**
+     * Attempt to submit an application transaction. Similar to
+     * {@link #submitTransaction(ConsensusTransactionImpl, boolean)} but with extra safeguards.
+     *
+     * @param transaction the transaction to submit
+     * @return true if the transaction passed all validity checks and was accepted by the consumer
+     */
+    public synchronized boolean submitApplicationTransaction(@NonNull final SwirldTransaction transaction) {
+        if (!healthy || platformStatus != PlatformStatus.ACTIVE) {
+            return false;
+        }
+
+        if (transaction == null) {
+            // FUTURE WORK: This really should throw, but to avoid changing existing API this will be changed later.
```

Comment:
I know that I often forget it myself, but can you open a issue for this

Reply:
There is currently an ongoing discussion as to how this can be best refactored. This entire class will almost certainly disappear as a result. Will open tickets when design is finalized.